### ISSUE TITLE
Support for connect go

### ISSUE THREAD

**Issue by acbeni** [grpc-ecosystem/go-grpc-middleware#622]
While working with another code generator ,[connect go](https://github.com/connectrpc/connect-go) i missed the rich middleware env provided by this lib, and had the idea of forking this repo to modify the code to be suitable for the [Interceptor](https://github.com/connectrpc/connect-go/blob/6accdb571faaa4425b621bdffba70633c5d9ab10/interceptor.go#L44-L48) interface in [connect go](https://github.com/connectrpc/connect-go). 

But before that i wanted to ask if its healthier to refactor the common parts, and try to make this library compatible with both interfaces (the second being the [grpc interceptor](https://github.com/grpc/grpc-go/blob/master/interceptor.go) from grpc go) in which case i can contribute with PRs. or you would rather keep focus on [grpc go](https://github.com/grpc/grpc-go) and leave forks for other implementations.

Thanks in advance 🙏

**Comment by johanbrandhorst** [comment]
I think it'd be great to have some way to adapt the middleware in this repo for use with Connect. The library is already pretty dynamic so it might be possible to write a simple adapter.

CC @bwplotka @akshayjshah

**Comment by bwplotka** [comment]
Makes sense. Would be nice to see how much work/code to maintain there is
with some PoC MVP, but otherwise would be keen to have it working for both.

Kind Regards,
Bartek Płotka ***@***.***)

**Comment by acbeni** [comment]
Thank you for your encouraging replies, one question do you want me to do a POC on a fork and post link here, or how do you want me to proceed?

**Comment by johanbrandhorst** [comment]
Nah just open a PR and we can take it from there

**Comment by acbeni** [comment]
Sounds good to me 👍
